integ-tests/src/main/java/com/google/base/TestBase.java

```diff
@@ -51,16 +51,26 @@
    * @return Date as a timestamp as Long.
    */
   public static long getTimestampByDate(String dateString) {
+    final String dateFormat = "yyyy-MM-dd HH:mm:ss.SSS";
+    return getTimestampByCustomDate(dateString, dateFormat);
+  }
+
+  /**
+   * @param dateString Date passed as String.
+   * @param dateFormat Custom date format as String.
+   * @return Date as a timestamp as Long.
+   */
+  public static long getTimestampByCustomDate(String dateString, String dateFormat) {
     if (dateString.equals("")) {
       return 0;
     } else {
       try {
-        SimpleDateFormat datetimeFormatter1 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
-        Date date = datetimeFormatter1.parse(dateString);
+        SimpleDateFormat datetimeFormatter = new SimpleDateFormat(dateFormat);
```

Comment:
This means you need to reconstruct the SimpleDateFormat on each call. I'd rather move this to a class level constant. I.e., change the signature of this method to take `String, dateString, SimpleDateFormat dateFormat`.

Speaking of `SimpleDateFormat`: While not deprecated, in modern Java, [DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html) is preferred because it fits better to the `java.time` API introduced in Java 8, but also because it is thread safe.

Reply:
I tackled the problem from a different angle.
I checked the extraction tool code and found out that there is a Jdbc method getTimestamp(), and it solves all questions in this date parsing situation.